lib/src/clojure_lsp/feature/completion.clj

```diff
@@ -207,64 +207,60 @@
-                                                      (f.add-missing-libspec/add-known-alias (symbol (str element-alias))
-                                                                                             (symbol (str element-to))
-                                                                                             db)
-                                                      r.transform/result)]
-                             (cond-> {:label label
-                                      :priority :required-alias
-                                      :kind :property
-                                      :detail detail}
-                               (seq require-edit) (assoc :additional-text-edits (mapv #(update % :range shared/->range) require-edit)))))))
-                  aliases))
+                           (q/ns-aliases db))]
+      (concat
+        (when (simple-ident? cursor-value)
+          (into []
+                (comp
+                  (filter (fn [element]
+                            (or
+                              (matches-fn (:alias element))
+                              (matches-fn (:to element)))))
+                  (map (fn [element]
+                         [(some-> element :alias name)
+                          (some-> element :to name)]))
+                  (distinct)
+                  (map (fn [[element-alias element-to]]
+                         (let [match-alias? (matches-fn element-alias)
+                               label (if match-alias?
+                                       element-alias
+                                       element-to)
+                               detail (if match-alias?
+                                        (str "alias to: " element-to)
+                                        (str ":as " element-alias))
+                               require-edit (some-> cursor-loc
+                                                    (f.add-missing-libspec/add-known-alias (symbol (str element-alias))
+                                                                                           (symbol (str element-to))
+                                                                                           db)
+                                                    r.transform/result)]
+                           (cond-> {:label label
+                                    :priority :required-alias
+                                    :kind :property
+                                    :detail detail}
+                             (seq require-edit) (assoc :additional-text-edits (mapv #(update % :range shared/->range) require-edit)))))))
+                aliases))
+        (let [alias-namespaces (->> aliases
+                                    (filter #(= (-> % :alias str) cursor-alias))
+                                    (map :to)
+                                    seq
+                                    set)]
```

Comment:
nice

Reply:
I still have a hard time reasoning about what this code does. Maybe I'll write myself a little note describing what it returns and when.